src/components/search/SearchBar.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useEffect } from "react";
+import { Input, Form, Button } from "reactstrap";
+import axios from "axios";
+import { useParams } from "react-router";
+import { useNavigate } from "react-router-dom";
+import OneCard from "../card/OneCard";
+
+const SearchBar = ({ headers }) => {
+  const [searchInput, setSearchInput] = useState("");
+  const [query, setQuery] = useState("");
+  const [queryResult, setQueryResult] = useState("");
+  const [searchEndpoint, setSearchEndpoint] = useState("");
+  const { header } = useParams();
+  const navigate = useNavigate();
+
+  let name = headers.map(({ name }) => {
```

Comment:
This is only used in `handleSearch` so move the variable declaration inside that function declaration